### PR TITLE
feat(race-engineer): pit-box count-in — distance countdown to your pit box (#600)

### LINKED ISSUE #600: feat(race-engineer): pit-box count-in — distance countdown to your pit box ("five… four… pit now")
## What would you like?

A new **Race Engineer voice callout** that counts the driver down to their **pit box** by distance while they're on pit road: **"five… four… three… two… one… pit now."** Each cue is spoken as the remaining distance to the box crosses a fixed mark, so the driver knows exactly when to brake and stop without overshooting the stall.

Distance marks (remaining distance to the box):

| Cue | Remaining distance |
|-----|--------------------|
| "five" | 120 m |
| "four" | 100 m |
| "three" | 80 m |
| "two" | 60 m |
| "one" | 40 m |
| "pit now" | 20 m |

Behaviour:

- **Trigger:** fires whenever the player is on pit road (`OnPitRoad === true`) and approaching their box. It is accepted that this also fires on drive-throughs and tracks that route through pit road — a future enhancement could optionally gate it to "only when pit service is requested".
- **Box location** comes from `sessionInfo.DriverInfo.DriverPitTrkPct` (the player's assigned stall as a 0–1 fraction of the lap), so the count-in works on the first stop of a session — no need to have visited the box before.
- Each mark is spoken **once** per pit-road visit; the set resets when the player leaves pit road so a second stop counts down again.
- If the player enters pit road already closer than 120 m (short pit lane / box near entry), only the marks still ahead are spoken.
- Once the box is passed (remaining distance wraps past the stall), no further marks fire.
- New per-callout opt-in toggle (default on), consistent with the other Race Engineer callouts.

## Use case

Overshooting your pit box in iRacing costs real time (you have to reverse or get waved around) and incurs penalties in some series. A spoken distance count-in lets the driver judge the stop by ear while keeping eyes on the box and the limiter, instead of guessing. It mirrors what a real race engineer / spotter does on the radio during a stop.

## Area

Action (existing) — **Pit Crew / Race Engineer** voice callouts (`com.iracedeck.sd.core.pit-crew`). This is a new callout family within the existing Race Engineer system, not a new action.

## Alternatives considered

- **Learn the box position by recording `LapDistPct` the first time `PlayerCarInPitStall` is true** — rejected: unnecessary now that `DriverInfo.DriverPitTrkPct` gives the box up front, and the learn-once approach can't count down on the very first stop.
- **Gate to "only when pit service requested" (`PitSvFlags`)** — deferred: keeps the first version simple; can be added later as an opt-in if drive-through false-fires prove annoying.
- **Configurable mark distances in the Property Inspector** — rejected for v1 (YAGNI); ship the fixed marks above with only an on/off toggle.
- **On-screen / key-icon countdown instead of voice** — rejected: the request is for the engineer to speak it; voice keeps the driver's eyes on the box.

## Additional context — implementation notes

This follows the end-to-end checklist in `.claude/rules/race-engineer-callouts.md` as a **new family** (`pit-box`). Naming per `.claude/rules/global-settings.md`.

### Distance computation

`remainingFraction = ((DriverPitTrkPct - LapDistPct) mod 1)`, then `remainingMeters = remainingFraction × trackLengthMeters` (reuse the cached `state.trackLengthMeters` from `resolveTrackLengthMeters`, `translator.ts:1039`). A mark fires when `remainingMeters` crosses from above to at/below its threshold. Treat the box as "passed" when `remainingFraction` jumps back toward a full lap, and stop. Skip silently when `DriverPitTrkPct` is missing/0 or `trackLengthMeters` is unknown.

### Bus event (`@iracedeck/event-bus`)

- New event `pitBox.countdown` in `packages/event-bus/src/event-catalog.ts`, payload `{ mark: "five" | "four" | "three" | "two" | "one" | "pit-now" }`. Export any new value type from `packages/event-bus/src/index.ts`.

### Translator + diff (`@iracedeck/sim-events-iracing`)

- New diff module `packages/sim-events-iracing/src/diff/pit-box-countdown.ts`, wired into `translator.ts` `handleTick` (reads `OnPitRoad`, `LapDistPct`, `DriverPitTrkPct`, `state.trackLengthMeters`).
- New `TranslatorState` fields in `state.ts` (e.g. `pitBoxMarksSpoken` for this visit, plus the player's box pct cache) — update **both** the type and `createInitialState()`. Reset the spoken set on `pitLane.exited` / when not on pit road.
- Tests: first-tick seeding, each threshold crossing fires once, no repeat while between marks, reset on leaving pit road, entering already-close (skips earlier marks), box-passed stops, missing `DriverPitTrkPct` stays silent.

### Voice clips (`@iracedeck/audio-assets`)

- New clip group `pit-box` in `configs/default.voice.json` (and every other voice file — `voice-parity.test.ts` enforces matching keys): entries `five-01`, `four-01`, `three-01`, `two-01`, `one-01`, `pit-now-01`.
- Generate: `pnpm --filter @iracedeck/audio-assets generate --group pit-box` then `generate:manifest`.

### Audio pools + scenario (`@iracedeck/audio-scenarios`)

- One pool per mark in `pools.ts` (`pit-box-five` … `pit-box-pit-now`).
- New scenario file `packages/audio-scenarios/src/catalog/pit-crew/pit-box.ts`, `family: "pit-box"` so successive marks preempt each other. Resolve the clip from `mark` (dynamic clip var via `engine.defineVar`, like the lap-time / race-start patterns) rather than six near-identical scenarios. Terse delivery — **no** `radio-open`/`radio-close` beep wrappers around each number (keeps the countdown crisp); priority `normal`.
- Wire into `catalog/pit-crew/index.ts`: family id type, `PIT_BOX_CALLOUT_SETTING_KEYS`, scenario-id map, `getPitBoxCalloutEnabled` param on `registerPitCrew` (before the master-gate param), wrap with `wrapWithMaster(wrapCalloutScenario(...))`.

### Opt-in (`@iracedeck/deck-core`)

- New Zod field `calloutEnabledPitBoxCountIn` in `global-settings.ts` using the standard boolean/string coercion, **default `true`**.
- Add the new key to the exhaustive `getGlobalSettings()` mock in `simhub-service.test.ts`.

### Property Inspector (`@iracedeck/iracing-actions`)

- Add a checkbox row to the Race Engineer Callouts accordion in `pit-crew.ejs` (`calloutEnabledPitBoxCountIn`, label e.g. "Pit-box count-in", `global default="true"`).

### Both plugins

- `iracing-plugin-stream-deck/src/plugin.ts` **and** `iracing-plugin-mirabox/src/plugin.ts`: pass a closure to `registerPitCrew` that **live-reads** the opt-in via `PIT_BOX_CALLOUT_SETTING_KEYS` (`!== false`). Keep the master-gate arg in its existing slot (the test fixtures call `registerPitCrew` positionally — add `undefined` at the new position in `register-pit-crew.test.ts`).

### Scenario harness (`@iracedeck/scenario-harness`)

- Add a shortcut button in `scenario-shortcuts.ts` to fire `pitBox.countdown` for each mark, and an entry in `event-names.ts` for the new event (compile-time completeness check requires it).

### Docs / website / skills / rules (must stay in sync — see `.claude/rules/build-and-commit.md`)

- Website (`@iracedeck/website`): Pit Crew / Race Engineer callouts feature list + any callout count.
- Action docs (`docs/`): Pit Crew action doc / Race Engineer callout listing.
- Skills: `iracedeck-actions` callout listing if it enumerates callouts.
- Rules: add a reference-implementation bullet to `.claude/rules/race-engineer-callouts.md`.

### Verify

`pnpm install && pnpm build && pnpm test && pnpm lint:fix && pnpm format:fix`. Manual: trigger each mark from the scenario harness (no iRacing needed), then in iRacing drive down pit road to your box and confirm the marks fire at the right distances and reset on a second stop. Toggle the PI opt-in mid-session to confirm the live-read silences future fires.

### PR DESCRIPTION
**Pull request by niklam**
## Related Issue

Fixes #600

## What changed?

Adds a **pit-box count-in** Race Engineer callout: as the driver drives down pit road toward their pit box, the engineer counts the remaining distance down — "five" (120 m), "four" (100 m), "three" (80 m), "two" (60 m), "one" (40 m), "pit now" (20 m) — so they know when to stop without overshooting the stall.

- **event-bus** — new `pitBox.countdown` event + `PitBoxMark` type.
- **sim-events-iracing** — `diffPitBoxCountdown` (band-selects the mark from the `LapDistPct`→box gap × cached track length; one mark per pit-road visit; resets off pit road; gated out in the stall) + `resolvePitBoxTrkPct` (reads `DriverInfo.DriverPitTrkPct`, so it works on the first stop of a session — no learn-once needed).
- **audio** — `pit-box` voice group (6 clips, generated) + pools + 6 per-mark scenarios mapping to a single `count-in` opt-in (`family: "pit-box"`, terse — no radio-beep frame).
- **deck-core** — `calloutEnabledPitBoxCountIn` global setting (default on).
- **Property Inspector** — "Pit Box" row in the Race Engineer Callouts accordion + live-read opt-in closures in **both** plugins.
- **scenario-harness** — 6 shortcut buttons + event template.
- **docs / website / rule** — feature sections + reference-implementation entry.

The count-in fires whenever you're on pit road approaching your box (including drive-throughs); it isn't tied to a pit-service request.

## How to test

**Scenario harness (no iRacing):** fire the **Pit Box** shortcuts (Five … Pit now) and confirm each mark plays its clip; fire two in quick succession to confirm same-family preemption.

**In iRacing:** drive down pit road to your box — confirm "five … pit now" fire at roughly 120 / 100 / 80 / 60 / 40 / 20 m remaining, that each mark fires once, and that a second stop counts down again. Toggle **Race Engineer Callouts → Pit Box** mid-session to confirm the live opt-in silences future fires without cutting an in-flight clip.

**Automated:** `pnpm build` (18/18) · `pnpm test` (4464/4464, incl. 12 new diff tests + 6 gating tests) · lint + format clean.

## Checklist

- [x] Linked to an approved issue
- [x] New code has unit tests
- [x] Changes registered in all applicable plugins (Stream Deck, Mirabox)
- [x] No unrelated changes included

## Summary by CodeRabbit

* **New Features**
  * Pit-box count-in callouts: Race Engineer speaks remaining distance to your pit box (“five”…“pit now”) once per pit-road visit; new runtime event published and voice clips added.
  * New Race Engineer Callouts toggle to enable/disable the pit-box count-in.

* **Documentation**
  * User guide and configuration docs for Pit-box count-in added.

* **Tests**
  * Unit tests and one-click scenario shortcuts added to validate countdown behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/niklam/iracedeck/pull/601?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/sim-events-iracing/src/diff/pit-box-countdown.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  const remainingFraction = (((boxTrkPct - lapDistPct) % 1) + 1) % 1;
+  const remainingMeters = remainingFraction * trackLengthMeters;
+
+  const mark = selectPitBoxMark(remainingMeters);
+
+  if (mark === null || state.pitBoxMarksSpoken.has(mark)) return;
+
+  state.pitBoxMarksSpoken.add(mark);
+  emit({ event: "pitBox.countdown", data: { mark } });
+}
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/sim-events-iracing/src/diff/pit-box-countdown.ts:159]: Good catch — fixed in 3d1bc39b.

On the first valid tick of a pit-road visit the diff now seeds every already-passed threshold into `pitBoxMarksSpoken` (gated by a new `state.pitBoxEntrySeeded` flag that's cleared when the car leaves pit road), so a mark only fires on a genuine downward crossing. Joining pit road at ~70 m now starts at **"two"** instead of speaking the just-passed **"three"**.

Added a dedicated regression test (`entering within range (threshold-crossing)` — asserts entry at 70 m yields `["two","one","pit-now"]` and never `"three"`) and updated the existing tests to enter above the first mark so each threshold is crossed from above. `pnpm build` / `pnpm test` (4464) / lint / format all green.

File: packages/sim-events-iracing/src/diff/pit-box-countdown.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  // First valid tick of this pit-road visit: seed every threshold the car is
+  // already past so only marks still AHEAD can fire (true threshold-crossing
+  // semantics). Without this, joining pit road mid-band would speak a
+  // just-passed number — e.g. entering at 70 m would say "three" (the 80 m
+  // mark). Gated on valid data rather than `OnPitRoad` so a visit that starts
+  // before the box position is known still seeds on the first tick it resolves.
+  if (!state.pitBoxEntrySeeded) {
+    for (const { mark, meters } of PIT_BOX_COUNTDOWN_MARKS) {
+      if (remainingMeters < meters) state.pitBoxMarksSpoken.add(mark);
+    }
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/sim-events-iracing/src/diff/pit-box-countdown.ts:148]: Applied in 5825b10c. Extracted a named `PIT_BOX_EPSILON_METERS` and made the entry seeding strict-symmetric with the inclusive band check — seeding now uses `remainingMeters < meters - PIT_BOX_EPSILON_METERS` against `selectPitBoxMark`'s `<= meters + PIT_BOX_EPSILON_METERS`, so a boundary value within epsilon is left fireable rather than dropped for the visit. Existing diff tests stay green (build/test/lint/format all pass).
